src/main/java/org/terasology/joshariasSurvival/systems/SheepShearingSystem.java

```diff
@@ -0,0 +1,78 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.joshariasSurvival.systems;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.engine.core.Time;
+import org.terasology.engine.entitySystem.entity.EntityManager;
+import org.terasology.engine.entitySystem.entity.EntityRef;
+import org.terasology.engine.entitySystem.event.ReceiveEvent;
+import org.terasology.engine.entitySystem.systems.BaseComponentSystem;
+import org.terasology.engine.entitySystem.systems.RegisterSystem;
+import org.terasology.engine.logic.characters.events.AttackEvent;
+import org.terasology.wildAnimals.event.AnimalSpawnEvent;
+import org.terasology.engine.logic.delay.DelayManager;
+import org.terasology.engine.logic.delay.PeriodicActionTriggeredEvent;
+import org.terasology.engine.logic.inventory.ItemComponent;
+import org.terasology.engine.registry.In;
+import org.terasology.joshariasSurvival.components.ShearableSheepComponent;
+
+
+import java.util.ArrayList;
+
+@RegisterSystem
+public class SheepShearingSystem extends BaseComponentSystem {
+    public static final int HITS_TO_SHEAR = 5;
+    public static final ArrayList<ItemComponent> SHEARING_ITEMS = null; //TODO yet to decide
+    public static final int HAIR_REGROWTH_TIME = 3*60*1000; // In ms
+    public static final String HAIR_REGROWTH_ACTION_ID = "Check hair regrowth";
+    private static final Logger logger = LoggerFactory.getLogger(SheepShearingSystem.class);
+
+    @In
+    protected Time time;
+
+    @In
+    private DelayManager delayManager;
+
+    @In
+    private EntityManager entityManager;
+
+    @ReceiveEvent
+    public void onAnimalSpawn(AnimalSpawnEvent event, EntityRef entityRef) {
+        if(entityRef.getParentPrefab().toString().contains("WildAnimals:sheep")) {
+            entityRef.addComponent(new ShearableSheepComponent());
+        }
+    }
+
+    @ReceiveEvent(components = {ShearableSheepComponent.class})
+    public void onAnimalSpawn(AttackEvent event, EntityRef entityRef) {
```

Comment:
nitpick: method name

(thought: It's taking me a while to get used to Terasology's event handlers where you have to look at the type of the method's _arguments,_ not the its name or its annotation, to know which event it responds to.)

Reply:
Yeah, having two methods called the same isn't really a good practise.